app/controllers/application_controller.rb

```diff
@@ -5,6 +5,83 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/create'
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if @user.save
+      session[:user_id] = @user.id
+      redirect '/tweets'
+    else
+      @message = "Oops, something didn't work.  Try again."
+      redirect "/signup"
+    end
+  end
+
+  get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).